pkg/workflows/management/pre_cluster_upgrade.go

```diff
@@ -0,0 +1,49 @@
+package management
+
+import (
+	"context"
+
+	"github.com/aws/eks-anywhere/pkg/logger"
+	"github.com/aws/eks-anywhere/pkg/task"
+	"github.com/aws/eks-anywhere/pkg/workflows"
+)
+
+type preClusterUpgrade struct{}
+
+// Run preClusterUpgrade implements steps to be performed before management cluster's upgrade.
+func (s *preClusterUpgrade) Run(ctx context.Context, commandContext *task.CommandContext) task.Task {
+	// Take best effort CAPI backup of workload cluster without filter.
+	// If that errors, then take CAPI backup filtering on only workload cluster.
```

Comment:
Does this comment mean we initially try to backup all CAPI components and if that fails then we try to backup only the CAPI components for the cluster being upgraded?

If that's what we are trying to do, then this doesn't seem accurate. I think this will create a backup of only the management cluster's CAPI components and not the workload clusters

```
commandContext.ClusterManager.BackupCAPI(ctx, commandContext.ManagementCluster, commandContext.ManagementClusterStateDir, commandContext.ManagementCluster.Name)
```